documentation/src/docs/watermarking.md

```diff
@@ -0,0 +1,121 @@
+
+Watermarking depends on optional features in Pravega writers and readers.
+* Pravega Writers must note time on each event (or use an optional feature that automatically timestamps events).
+* Pravega Readers request a time window. The lower bound of that window is the watermark. The application can then act on or process events before the watermark, knowing that the data is complete up to the last watermark.   
+
+Note: The watermark is a best ability mark. There is always the possibility that a reader is unusually delayed and recovers and processes an event that falls earlier than a watermark (earlier than a closed window of time). In such a case, it is up to the application to determine a recourse.
+
+Uses of watermarking include applications that require a time bounded   set of metrics to make a calculation or make a decision. Transactional applications are sometimes time-bound.
+
+The Flink API offers watermarking features. The Pravega watermarking feature mirrors the Flink functionality,   enabling an easier coding experience for integrating Pravega and Flink applications.
+
+## More information
+
+Read more about the Flink watermarking feature and example use cases <here>.
+Read more about the Pravega watermarking design and usage in the Pravega blog <here>.
+
+## About time
+
+Watermarking is an optional feature in Pravega. If implemented, the concept of time is defined by the application. The APIs provide  the means for writers to note the time  and for readers to request the watermarked window of time, but the definition of  time itself is  controlled by the application.
+
+The time may indicate the time the data was written to the stream (ingestion time).  Time may alternatively be equated to event time (the time the event occurred).  Time can denote wall clock time, some other traditional time, or an arbitrary value, like 6.    
+
+Time has two requirements:
+
+* Time is a variable of type long.  
+* Time must increase in value (not decrease). That is, if time is 6, then a time of 5 is earlier, and a time of 7 is later.
+
+## How Pravega Writers note the time
+
+There are three ways for Pravega Writers to attach time to events.
+* Explicitly note the time  
+* Note time on transaction commit
+* Automatically note wall clock time
+
+### Explicitly note time
+
+The following writer API method notes time.
+
+`writer.noteTime(long timestamp);`
+
+You may call the method after every event write, or use it periodically to indicate  the time below which all events are written.
```

Comment:
Here and a lot of other places there are spaces instead of one randomly in the middle of the line.